integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala

```diff
@@ -61,241 +62,167 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier])
-        val entry = cacheIterator.next()
-        val cache = entry.getValue
-
-        if (cache.isInstanceOf[BlockletDataMapIndexWrapper]) {
-          // index
-          val indexPath = entry.getKey.replace(CarbonCommonConstants.WINDOWS_FILE_SEPARATOR,
-            CarbonCommonConstants.FILE_SEPARATOR)
-          val pathEntry = datamapSize.filter(entry => indexPath.startsWith(entry._1))
-          if(pathEntry.nonEmpty) {
-            val (path, size) = pathEntry.iterator.next()
-            datamapSize.put(path, size + cache.getMemorySize)
-          }
-          if(indexPath.startsWith(tablePath)) {
-            numIndexFilesCached += 1
-          }
-        } else if (cache.isInstanceOf[BloomCacheKeyValue.CacheValue]) {
-          // bloom datamap
-          val shardPath = entry.getKey.replace(CarbonCommonConstants.WINDOWS_FILE_SEPARATOR,
-            CarbonCommonConstants.FILE_SEPARATOR)
-          val pathEntry = datamapSize.filter(entry => shardPath.contains(entry._1))
-          if(pathEntry.nonEmpty) {
-            val (path, size) = pathEntry.iterator.next()
-            datamapSize.put(path, size + cache.getMemorySize)
-          }
-        } else if (cache.isInstanceOf[AbstractColumnDictionaryInfo]) {
-          // dictionary
-          val dictId = dictIds.find(id => entry.getKey.startsWith(id))
-          if (dictId.isDefined) {
-            dictSize = dictSize + cache.getMemorySize
-          }
-        }
+  def getTableCache(sparkSession: SparkSession, carbonTable: CarbonTable): Seq[Row] = {
+    val cache = CacheProvider.getInstance().getCarbonCache
+    val tablePath = carbonTable.getTablePath + CarbonCommonConstants.FILE_SEPARATOR
+    val showTableCacheEvent = ShowTableCacheEvent(carbonTable, sparkSession, internalCall)
+    val operationContext = new OperationContext
+    // datamapName -> (datamapProviderName, indexSize, datamapSize)
+    val currentTableSizeMap = scala.collection.mutable.Map[String, (String, String, Long, Long)]()
+    operationContext.setProperty(carbonTable.getTableUniqueName, currentTableSizeMap)
+    OperationListenerBus.getInstance.fireEvent(showTableCacheEvent, operationContext)
+
+    // Index files
+    val allIndexFiles: List[String] = CacheUtil.getAllIndexFiles(carbonTable)
+    val indexFilesInCache: List[String] = allIndexFiles.filter {
+      indexFile =>
+        cache.get(indexFile) != null
+    }
+    val sizeOfIndexFilesInCache: Long = indexFilesInCache.map {
+      indexFile =>
+        cache.get(indexFile).getMemorySize
+    }.sum
+
+    // Dictionary IDs
+    val dictIDs: List[String] = carbonTable.getAllDimensions.asScala
+      .filter(_.isGlobalDictionaryEncoding)
+      .map(_.getColumnId).toList
+    val dictKeys: List[String] =
+      dictIDs.map {
+        dictID =>
+          CacheUtil.getLruCacheKey(dictID, CacheType.FORWARD_DICTIONARY)
+      } ++
+      dictIDs.map {
+        dictID =>
+          CacheUtil.getLruCacheKey(dictID, CacheType.REVERSE_DICTIONARY)
```

Comment:
combine reverse and forward into one map

Reply:
Done.